designs/v1beta1-full-changelist.md

```diff
@@ -0,0 +1,200 @@
+
+This gives the following naming to API types within the Karpenter project
+
+1. `karpenter.sh/NodePool`
+2. `karpenter.sh/NodeClaim`
+3. `compute.k8s.aws/NodeClass`
+
+### Remove Validation/Mutating Webhooks in favor of CEL (Common Expression Language)
+
+The Karpenter maintainer team has seen an increase in the number of issues related to its webhooks ([#4415](https://github.com/aws/karpenter/issues/4415), [#3598](https://github.com/aws/karpenter/issues/3598), [#2902](https://github.com/aws/karpenter/issues/2902), [#4154](https://github.com/aws/karpenter/issues/4154), [#4106](https://github.com/aws/karpenter/issues/4016), [#3224](https://github.com/aws/karpenter/issues/3224), [#1729](https://github.com/aws/karpenter/issues/1729), ...) which lead us to believe that we should look for alternatives/ways to remove these webhooks from the project.
+
+Kubernetes 1.23 introduced the `CustomResourceValidationExpressions` in alpha, followed by graduating the feature to beta in 1.25. This feature introduces the ability to write CRD validation expressions directly in the CRD OpenAPISpec without any need for validating webhooks to do custom validation. EKS supports CEL starting in Kubernetes version 1.25.
+
+Karpenter v1beta1 will introduce CEL into the CRD OpenAPISpec while maintaining the webhooks until support for EKS versions <= 1.24 is dropped. At this point, we will drop support for the webhooks and rely solely on CEL for validation.
+
+### Label Changes
+
+### `karpenter.sh/do-not-evict` →  `karpenter.sh/do-not-disrupt`
+
+Karpenter validates disruption across NodeClaims and determines which NodeClaims/Nodes it is allowed to disrupt as part of the deprovisioning flow. While eviction is part of the termination process, it’s more accurate to say that the `karpenter.sh/do-not-evict` annotation actually prevents Karpenter’s disruption of the NodeClaim/Node rather than the eviction of it.
+
+###  `karpenter.sh/do-not-consolidate`  → `karpenter.sh/do-not-disrupt`
+
+Karpenter currently surfaces the `karpenter.sh/do-not-consolidate` annotation to block consolidation actions against individual nodes without having to make changes to the owning provisioner. We have found this is useful for users that have one-off scenarios for blocking consolidation, including debugging failures on nodes. 
+
+While this feature is useful for consolidation, it should be expanded out to all disruption mechanisms, so that we have both pod-level and node-level control to block disruption using the `karpenter.sh/do-not-disrupt` annotation.
```

Comment:
Is it possible to add this annotation also at the Provisioner/ NodePool level? This would allow an entire pool of nodes to be temporarily frozen for certain operation.

Reply:
Hey @garvinp-stripe, there's a separate effort to include time-based controls that come with a `maxDisruptions` type field. The end state in conjunction with this proposal is that each TTL field can be enabled/disabled so that users can define which types of deprovisioning they want for a Provisioner/NodePool, and then in the same Provisioner/NodePool, users can define a schedule that determine how many nodes can be deprovisioned at a time, irrespective of the method. You would be able to set `maxDisruptions:0` to effectively freeze the Provisioner/NodePool.

Reply:
Make sense, thanks